apps/shipments/management/commands/replicate_shipments.py

```diff
@@ -32,24 +34,31 @@ def add_arguments(self, parser):
         )
 
     def _replicate_shipment(self, shipment):
-        self.successfull_shipments.append(shipment.id)
         for async_job in AsyncJob.objects.filter(shipment=shipment):
             try:
-                eth_action = EthAction.objects.filter(async_job=async_job).first()
+                message = Message.objects.filter(async_job=async_job, type=MessageType.ETH_TRANSACTION).last()
+                if not message or 'cumulativeGasUsed' not in message.body or message.body['cumulativeGasUsed'] == '0':
+                    continue
+                logger.info(f'Performing action for shipment: {shipment.id}')
+                AsyncTask(async_job.id).rerun()
+                eth_action = EthAction.objects.filter(async_job=async_job).last()
```

Comment:
I wanted to wait until afterwards in case there was an exception that caused the `AsyncJob` to not be run.
You're right, they should all be deleted. The reason this was done with the `last` tag was after testing locally when the action was retrieved after creating the new one, and shouldn't be necessary.